src/searchModal.js

```diff
@@ -26,38 +26,59 @@ import component from 'ui/component';
 import 'ui/modal';
 import 'ui/datatable';
 import store from 'core/store';
-
+import resourceSelectorFactory from 'ui/resource/selector';
+import request from 'core/dataProvider/request';
+import urlUtil from 'util/url';
 /**
  * Creates a searchModal instance
  *
  * @param {object} config
  * @param {object} config.renderTo - DOM element where component will be rendered to
- * @param {string} config.query - search query to be set on component creation
+ * @param {string} config.criterias - Search criterias to be set on component creation
  * @param {boolean} config.searchOnInit - if init search must be triggered or not (stored results are used instead)
  * @param {string} config.url - search endpoint to be set on datatable
+ * @param {string} config.rootClassUri - Uri for the root class of current context, required to init the class filter
+ * @param {string} config.testMocks - mocks to be used for testing to avoid ajax requests
  * @returns {searchModal}
  */
 export default function searchModalFactory(config) {
     const defaults = {
         renderTo: 'body',
-        query: '',
-        searchOnInit: true
+        criterias: {},
+        searchOnInit: true,
+        testMocks: {}
     };
+    /*
+     * Not all context resources are indexed on DB so on those contexts
+     * BE does not support search by class, so we need to hide and
+     * disable this functionality.
+     */
+    const classFilterIsEnabled = config.rootClassUri !== undefined;
```

Comment:
```suggestion
    const isClassFilterEnabled = config.rootClassUri !== undefined;
```

Reply:
After discussion with BA class filter must never be disabled so this constant and every logic related has been removed.